crates/sparrow-main/tests/e2e/basic_error_tests.rs

```diff
@@ -380,19 +380,17 @@ async fn test_incompatible_actual_types_error() {
     message: 1 errors in Fenl statements; see diagnostics
     fenl_diagnostics:
       - severity: error
-        code: E0010
-        message: Invalid argument type(s)
+        code: E0015
+        message: Incompatible argument types
         formatted:
-          - "error[E0010]: Invalid argument type(s)"
+          - "error[E0015]: Incompatible argument types"
           - "  --> Query:1:16"
           - "  |"
           - "1 | { m: Strings.s + Strings.n }"
           - "  |      --------- ^ --------- Type: i64"
           - "  |      |         |  "
-          - "  |      |         Invalid types for call to 'add'"
+          - "  |      |         Incompatible types for call to 'add'"
```

Comment:
This change is _not_ as desirable in this context, perhaps?

Reply:
Seems OK? But why did we need a new "incompatible" when we had "invalid"? Does invalid still show up ever?

Reply:
We were conflating two separate scenarios to the `Invalid` error 
1. When two types for a type class did not have a least upper bound 
2. When the least upper bound from step 1 could not be promoted to a type that satisfies the type class. 

So I had the first instead return `Incompatible`, attempting to imply the argument types were incompatible with each other (because they do not have a least upper bound).

Reply:
Easy to fix up if it ends up being confusing, so going to resolve and merge this.